---
title: 'R Notebook for preprocessing GEO methylation profiling by array datasets (Illumina 27k+) for estimating epigenetic age with AltumAge or the 2013 Horvath epigenetic clock'
output: html_notebook
---


Variables, filenames:
```{r}
geo = 'GSE167202'
gpl = 'GPL8490'

fname = 'covid'
unprocessed_filename = paste('data/', fname, geo,'_unprocessed.csv', sep='')
meta_filename = paste('data/', fname, geo,'_meta.csv', sep='')
normalized_filename = paste('data/', fname, geo,'_normalized.csv', sep='')
horvath_filename = paste('data/', fname, geo,'_horvath.csv', sep='')
```
Dataset citation: 
Konigsberg IR, Barnes B, Campbell M, Davidson E, Zhen Y, Pallisard O, Boorgula M, Cox C, Nandy D, Seal S, Crooks K. Host methylation predicts SARS-CoV-2 infection and clinical outcome.

```{r}
library(GEOquery)
library(minfi)
library(dplyr)
library(impute)
library(data.table)
```

```{r}
gset <- getGEO(geo, GSEMatrix =TRUE, getGPL=FALSE)
if (length(gset) > 1) idx <- grep(gpl, attr(gset, "names")) else idx <- 1
gset <- gset[[idx]]
# ex <- exprs(gset) # empty for this gset
# dat0 = ex
#write.table(dat0, unprocessed_filename, row.names=T, sep=',') 

meta <- pData(gset)
write.table(meta, meta_filename, row.names=T, sep=',' ) 
```

```{r}
# Downloaded GSE167202_RAW.tar directly from https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE167202
cpgs27k = fread('HorvathClock/datMiniAnnotation27k.csv')$Name

idatFiles <- list.files('GSE167202_RAW', pattern = "idat.gz$", full = TRUE)
sapply(idatFiles, gunzip, overwrite = TRUE)

targets <- paste('GSE167202_RAW/',rownames(meta),'_', meta$description.1,'_noid',sep='')
targets_frame <- data.frame(Sample_Name=rownames(meta), Basename=targets)


for (i in 1:length(targets_frame[[1]])){
  rgs <- read.metharray.exp(targets = targets_frame[i,])
  rgs@annotation=c(array="IlluminaHumanMethylationEPIC",annotation="ilm10b2.hg19")
  rgs <-subsetByLoci(rgs, includeLoci=cpgs27k)
  if (i == 1){betas <-getBeta(rgs, type="Illumina")}
  else{betas <- cbind(betas, getBeta(rgs, type="Illumina"))}
  print(i)
}

betas_dataframe <- data.frame(betas)
betas_dataframe <- cbind('v1'=rownames(betas_dataframe), betas_dataframe)
fwrite(betas_dataframe, unprocessed_filename, row.names=F, sep=',') 
```

```{r}
# Downloaded GSE167202_ProcessedBetaValues.txt.gz from https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE167202 - first tried it out as is, and then did normalization with this file to see if re-normalizing could work for age estimation (it didn't)

# unprocessed = fread('GSE167202_ProcessedBetaValues.txt')
# unprocessed = fread()
# dat0 <- select(unprocessed,-2:-22)
# colnames(dat0)[-1] <- colnames(ex)
# match0=match(unlist(altum_cpgs) , dat0[[1]])
# match0
# dat0 <- dat0[match0]
# fwrite(dat0, unprocessed_filename, row.names=F, sep=',') 
```



The code and comments in the cells below are from Steve Horvath's "Estimate DNA methylation age with R tutorial" (with only some slight modifications): https://horvath.genetics.ucla.edu/html/dnamage/TUTORIAL1.pdf
Method citation: Horvath S (2013) DNA methylation age of human tissues and cell types. Genome Biology

```{r}
probeAnnotation21kdatMethUsed=read.csv("HorvathClock/probeAnnotation21kdatMethUsed.csv")
probeAnnotation27k=read.csv("HorvathClock/datMiniAnnotation27k.csv")
datClock=read.csv("HorvathClock/AdditionalFile3.csv")

unprocessed = fread(unprocessed_filename)
dat0 <- data.frame(unprocessed)
rownames(dat0) <- dat0[[1]]
dat0 = dat0[,-1]

nSamples=dim(dat0)[[2]]
nProbes= dim(dat0)[[1]] 
```

```{r}
XchromosomalCpGs=as.character(probeAnnotation27k$Name[probeAnnotation27k$Chr=="X"]) 
selectXchromosome=is.element(rownames(dat0), XchromosomalCpGs ) 
selectXchromosome[is.na(selectXchromosome)]=FALSE
meanXchromosome=rep(NA, nSamples) 

if (   sum(selectXchromosome) >=500 )  { 
  meanXchromosome= as.numeric(apply(as.matrix(dat0[selectXchromosome,-1]),2,mean,na.rm=TRUE)) }
if (  sum(is.na(meanXchromosome)) >0 ) { 
paste( "Comment: There are lots of missing values for X chromosomal probes for some of the samples. This is not a problem when it comes to estimating age but I cannot predict the gender of these samples." )  } 
```

```{r}
match1=match(probeAnnotation21kdatMethUsed$Name , rownames(dat0))

if  ( sum( is.na(match1))>0 ) {  
missingProbes= probeAnnotation21kdatMethUsed$Name[!is.element( probeAnnotation21kdatMethUsed$Name , rownames(dat0))]}
match1=match(probeAnnotation21kdatMethUsed$Name , rownames(dat0))

if  ( sum( is.na(match1))>0 ) stop(paste(sum( is.na(match1)), "CpG probes cannot be matched"))

dat1= dat0[match1,] 
asnumeric1=function(x) {as.numeric(as.character(x))} 
rows <- rownames(dat1)
dat1=apply(as.matrix(dat1),2,asnumeric1)
rownames(dat1) = rows
```

```{r}
fastImputation=FALSE
#STEP 1: DEFINE QUALITY METRICS

meanMethBySample =as.numeric(apply(as.matrix(dat1),2,mean,na.rm=TRUE))
minMethBySample   =as.numeric(apply(as.matrix(dat1),2,min,na.rm=TRUE))
maxMethBySample  =as.numeric(apply(as.matrix(dat1),2,max,na.rm=TRUE))

datMethUsed= t(dat1)

noMissingPerSample=apply(as.matrix(is.na(datMethUsed)),1,sum)
table(noMissingPerSample)

#STEP 2: Imputing 

if (! fastImputation & nSamples>1 & max(noMissingPerSample,na.rm=TRUE)<3000 ){

# run the following code if there is at least one missing
if ( max(noMissingPerSample,na.rm=TRUE)>0 ){
dimnames1=dimnames(datMethUsed)
datMethUsed= data.frame(t(impute.knn(t(datMethUsed))$data))
dimnames(datMethUsed)=dimnames1
} # end of if
} # end of if (! fastImputation )

if ( max(noMissingPerSample,na.rm=TRUE)>=3000 ) fastImputation=TRUE


if ( fastImputation | nSamples==1 ){
noMissingPerSample=apply(as.matrix(is.na(datMethUsed)),1,sum)
table(noMissingPerSample)
if ( max(noMissingPerSample,na.rm=TRUE)>0 & max(noMissingPerSample,na.rm=TRUE) >= 3000 ) {normalizeData=FALSE}

# run the following code if there is at least one missing
if ( max(noMissingPerSample,na.rm=TRUE)>0 & max(noMissingPerSample,na.rm=TRUE) < 3000 ){
dimnames1=dimnames(datMethUsed)
for (i in which(noMissingPerSample>0) ){
selectMissing1=is.na(datMethUsed[i,])
datMethUsed[i,selectMissing1] = as.numeric(probeAnnotation21kdatMethUsed$goldstandard2[selectMissing1])
} # end of for loop
dimnames(datMethUsed)=dimnames1
} # end of if
} # end of if (! fastImputation )

```
```{r}
# STEP 3: Data normalization (each sample requires about 8 seconds). It would be straightforward to parallelize this operation.

source("NORMALIZATION_horvath.R") # NORMALIZATION.R file from https://horvath.genetics.ucla.edu/html/dnamage/

normalizeData = TRUE

if (normalizeData ){
datMethUsedNormalized=BMIQcalibration(datM=datMethUsed,goldstandard.beta= probeAnnotation21kdatMethUsed$goldstandard2,plots=FALSE)
}
if (!normalizeData ){ datMethUsedNormalized=datMethUsed }
rm(datMethUsed); gc()

datMethUsedNormalized <- cbind(ID=rownames(datMethUsedNormalized), datMethUsedNormalized)

write.table(datMethUsedNormalized,normalized_filename, row.names=F, sep="," ) 
```

The cell below for estimating epigenetic age with the 2013 Horvath cpgs won't work with the GSE167202 dataset (not enough matching cpgs):
```{r}
#STEP 4: Predict age and create a data frame for the output (referred to as datout)
trafo= function(x,adult.age=20) { x=(x+1)/(1+adult.age); y=ifelse(x<=1, log( x),x-1);y }
anti.trafo= function(x,adult.age=20) { ifelse(x<0, (1+adult.age)*exp(x)-1, (1+adult.age)*x+adult.age) }


selectCpGsClock=is.element(colnames(datMethUsedNormalized), as.character(datClock$CpGmarker[-1]))

if ( sum( selectCpGsClock) < dim(datClock)[[1]]-1 ) {stop("The CpGs listed in column 1 of the input data did not contain the CpGs needed for calculating DNAm age. Make sure to input cg numbers such as cg00075967.")}
if ( sum( selectCpGsClock) > dim(datClock)[[1]]-1 ) {stop("ERROR: The CpGs listed in column 1 of the input data contain duplicate CpGs. Each row should report only one unique CpG marker (cg number).")}
if (nSamples>1 ) {
datMethClock0=data.frame(datMethUsedNormalized[,selectCpGsClock])
datMethClock= data.frame(datMethClock0[ as.character(datClock$CpGmarker[-1])])
dim(datMethClock)

predictedAge=as.numeric(anti.trafo(datClock$CoefficientTraining[1]+as.matrix(datMethClock)%*% as.numeric(datClock$CoefficientTraining[-1])))
} # end of if


Comment=ifelse ( predictedAge <0, "Negative DNAm age.", ifelse ( predictedAge >100, "Old DNAm age.", rep("",length(predictedAge))))

Comment[is.na(predictedAge)]="Age prediction was not possible. "

if ( sum( selectCpGsClock) < dim(datClock)[[1]]-1 ) {
Comment=rep("ERROR: The CpGs listed in column 1 of the input data did not contain the CpGs needed for calculating DNAm age. Make sure to input cg numbers such as cg00075967.",length(predictedAge) )}


if ( sum( selectCpGsClock) > dim(datClock)[[1]]-1 ) {
Comment=rep("ERROR: The CpGs listed in column 1 of the input data contain duplicate CpGs. Each row should report only one unique CpG marker (cg number).",length(predictedAge) )}


restSamples=-minMethBySample>0.05 | maxMethBySample>1.05;
restSamples[is.na(restSamples)]=FALSE
lab1="MAJOR WARNING: Probably you did not input beta values since either minMethBySample<-0.05 or maxMethBySample>1.05.";Comment[restSamples]= paste(Comment[restSamples],lab1)

restSamples= noMissingPerSample >0 & noMissingPerSample <=100;lab1="WARNING: Some beta values were missing, see noMissingPerSample."; Comment[restSamples]= paste(Comment[restSamples],lab1)
restSamples= noMissingPerSample >3000;lab1="MAJOR WARNING: More than 3k missing values!!"; Comment[restSamples]= paste(Comment[restSamples],lab1)

restSamples= noMissingPerSample >100 & noMissingPerSample <=3000 ;lab1="MAJOR WARNING: noMissingPerSample>100"
Comment[restSamples]= paste(Comment[restSamples],lab1)
restSamples=meanMethBySample>.35;
restSamples[is.na(restSamples)]=FALSE
lab1="Warning: meanMethBySample is >0.35";Comment[restSamples]= paste(Comment[restSamples],lab1)
restSamples=meanMethBySample<.25;
restSamples[is.na(restSamples)]=FALSE; lab1="Warning: meanMethBySample is <0.25"
Comment[restSamples]= paste(Comment[restSamples],lab1)
datout=data.frame(SampleID=colnames(dat1), DNAmAge=predictedAge, Comment, noMissingPerSample,meanMethBySample, minMethBySample, maxMethBySample)


if ( !is.null( meanXchromosome) ){  

if ( length( meanXchromosome)==dim(datout)[[1]] ){
predictedGender=ifelse(meanXchromosome>.4,"female",
ifelse(meanXchromosome<.38,"male","Unsure"))
datout=data.frame(datout,predictedGender=predictedGender,meanXchromosome=meanXchromosome)

} # end of if 

 } # end of if


write.table(datout,horvath_filename, row.names=F, sep=',' ) 
```

